syntax = "proto3";

package face_recognition;

service FaceRecognition {
  rpc SendImages (ImageRequest) returns (ImageResponse);
  rpc GetResults (ResultRequest) returns (ResultResponse);
}

message ImageRequest {
  repeated bytes images = 1;
  repeated string labels = 2;
}

message ImageResponse {
  bool success = 1;
  string message = 2;
}

message ResultRequest {
}

message CameraFrames {
  int32 camera_index = 1;
  repeated bytes frames = 2;
}

message ResultResponse {
  repeated CameraFrames camera_frames = 1;
  repeated string recognized_labels = 2;
}